compiler/ghc/default.nix

```diff
@@ -185,17 +189,43 @@ let
 
   targetCC = builtins.head toolsForTarget;
 
+  useHadrian = builtins.compareVersions ghc-version "9.4" >= 0;
+
+  inherit ((buildPackages.haskell-nix.cabalProject {
+      compiler-nix-name = "ghc8107";
+      src = haskell-nix.haskellLib.cleanSourceWith {
+        src = buildPackages.srcOnly {
+          name = "hadrian";
+          inherit src;
+          patches = ghc-patches;
+        };
+        subDir = "hadrian";
+      };
+    }).hsPkgs.hadrian.components.exes) hadrian;
+
+  hadrianArgs = "--flavour=${
+        "default"
+          + lib.optionalString (!enableShared) "+no_dynamic_ghc"
+          + lib.optionalString useLLVM "+llvm"
+      } --docs=no-sphinx -j -V";
+
+  rootDir =
+    if (haskell-nix.haskellLib.isCrossTarget || targetPlatform.isMusl) && useHadrian
+      then ""
+      else "lib/${targetPrefix}ghc-${ghc-version}/";
+  libDir =
+    if (haskell-nix.haskellLib.isCrossTarget || targetPlatform.isMusl) && useHadrian
+      then "lib"
+      else "lib/${targetPrefix}ghc-${ghc-version}" + lib.optionalString (useHadrian) "/lib";
```

Comment:
This should have a comment, why are rootDir and libDir empty with Hadrian? _and_ for crossTargets?

Reply:
Comment added and I include a `installStage1` variable to make it clearer why there is a difference.